app/models/student.rb

```diff
@@ -156,8 +154,23 @@ def submission_for(code_review)
     submissions.find { |submission| submission.code_review_id == code_review.id }
   end
 
+  def pairs(course=nil)
+    selected_attendance_records = course ? attendance_records.where("date between ? and ?", course.try(:start_date), course.try(:end_date)) : attendance_records
```

Comment:
What happens if `start_date` or `end_date` is `nil`?

Reply:
Courses always have a start_time and end_time (now calculated through class_times rather than hardcoded). The problem was specs. 

This is actually just because when I was adding class_times recently I didn't feel like I had the time to go through and fix all the specs that have courses for purposes other than testing class times. But actually I added in a stub for start_time and end_time to be used by tests where I'm not specifically testing those things, so I should be able to delete that now.